File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
            vector<int> answer(2, 0);
        vector<int> num;
        for (auto q : s)
        {
            num.push_back(q - '0');
        }
        int cnt = 0;
        while (num.size()!= 1)
        {

            for (auto n : num)
            {
                if (n == 1) cnt++;
                else answer[1]++;
            }
            num.clear();
            while (cnt != 0)
            {
                num.push_back(cnt%2);
                cnt /= 2;
            }
            answer[0]++;
        }

    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cs
using System;
using System.Linq;
public class Solution {
    public int[] solution(string s) {
        int[] answer = new int[2] { 0, 0 };

        while(s.Length != 1)
        {
            Console.WriteLine("s = " + s);

            int cnt = s.Where(x => x == '0').Count();
            int cnt2 = s.Where(x => x == '1').Count();

            answer[1] += cnt;
            s = Convert.ToString(cnt2, 2);

            answer[0]++;
        }
        return answer;
    }
}